### PR TITLE
chore: add plugin skeleton

### PR DESCRIPTION
**Pull request by kneal**

### CODE REVIEW

File: cmd/vela-docker/build.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+package main
+
+import "github.com/sirupsen/logrus"
+
+// Build represents the plugin configuration for build information.
+type Build struct {
+}
+
+// Validate verifies the Build is properly configured.
+func (b *Build) Validate() error {
```
[the remark this answers is not in the document]
Comment (kneal) [reply cmd/vela-docker/build.go:14]: Ignore will be implemented in future PR.

File: cmd/vela-docker/config.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// Config holds input parameters for the plugin
+type Config struct {
+}
+
+// Validate verifies the Config is properly configured.
+func (c *Config) Validate() error {
```
[the remark this answers is not in the document]
Comment (kneal) [reply cmd/vela-docker/config.go:16]: Ignore will be implemented in future PR.

File: cmd/vela-docker/plugin.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+// Plugin represents the configuration loaded for the plugin.
+type Plugin struct {
+	// build arguments loaded for the plugin
+	Build *Build
+}
+
+// Command formats and outputs the command necessary for
+// Docker to build and publish a Docker image.
+func (p *Plugin) Command() *exec.Cmd {
```
[the remark this answers is not in the document]
Comment (kneal) [reply cmd/vela-docker/plugin.go:21]: Ignore will be implemented in future PR.

File: cmd/vela-docker/config.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// Config holds input parameters for the plugin
+type Config struct {
```
[the remark this answers is not in the document]
Comment (kneal) [reply cmd/vela-docker/config.go:12]: Ignore will be implemented in future PR.

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**